spirv_common.hpp

```diff
@@ -296,6 +296,8 @@ struct SPIRType : IVariant
 		bool depth;
 		bool arrayed;
 		bool ms;
+		bool is_read = false;
```

Comment:
Hm, so to be clear, does this read/write difference only concern image2D (image load/store), or sampler2D as well? Making all sampler2D writeable would certainly be a really bad idea, but I don't see the big problem if it's image2D only, but there might be issues in Metal I don't know about. For sampler2D, this is known by the type system (sampled == 2), so sampler2D can always be forced to read-only.

I'm fine with keeping things as it is, but please add a comment in the SPIRType structure that this is a temporary workaround for Metal.